tests/openvino/native/quantization/test_weights_compression_statistics_caching.py

```diff
@@ -0,0 +1,160 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from copy import deepcopy
+from functools import partial
+from itertools import product
+
+import datasets
+import openvino as ov
+from optimum.intel.openvino import OVModelForCausalLM
+from transformers import AutoTokenizer
+
+import nncf
+from nncf.quantization.advanced_parameters import AdvancedCompressionParameters
+from nncf.scopes import IgnoredScope
+
+
+def create_transform_fn(model, tokenizer):
+    def transform_fn(data, model=model, tokenizer=tokenizer):
+        tokenized_text = tokenizer(data["text"], return_tensors="np")
+        input_ids = tokenized_text["input_ids"]
+
+        inputs = {"input_ids": input_ids, "attention_mask": tokenized_text["attention_mask"]}
+
+        batch_size = input_ids.shape[0]
+        if hasattr(model, "key_value_input_names"):
+            for input_name in model.key_value_input_names:
+                model_inputs = model.model.input(input_name)
+                shape = model_inputs.get_partial_shape()
+                shape[0] = batch_size
+                if shape[2].is_dynamic:
+                    shape[2] = 0
+                else:
+                    shape[1] = 0
+                inputs[input_name] = ov.Tensor(model_inputs.get_element_type(), shape.get_shape())
+
+        return inputs
+
+    return transform_fn
+
+
+def test_weight_compression_statistics_caching_opt_125m(tmp_path, mocker):
+    """
+    Evaluates the weight compression process for the 'facebook/opt-125m' model,
+    specifically focusing on validating the statistics caching mechanism. The test ensures
+    that the caching mechanism behaves correctly by:
+
+    1. Collecting statistics once during the compression process.
+    2. Loading statistics multiple times, corresponding to the number of different algorithm configurations tested.
+
+    The test iterates over various combinations of compression parameters, including:
+
+    - AWQ
+    - Group size for quantization
+    - Compression ratios
+    - Sensitivity metrics (e.g., Hessian Input Activation)
+    - GPTQ
+    - Scale estimation
+    - Ignored scope
+    """
+    from nncf.openvino.statistics.aggregator import OVStatisticsAggregator
+
+    collect_statistics_spy = mocker.spy(OVStatisticsAggregator, "collect_statistics")
+    load_statistics_from_dir_spy = mocker.spy(OVStatisticsAggregator, "load_statistics_from_dir")
+    dump_statistics_spy = mocker.spy(OVStatisticsAggregator, "dump_statistics")
+
+    # Constant Parameters
+    subset_size = 4
+    mode = nncf.CompressWeightsMode.INT4_ASYM
+
+    # Compression Parameters
+    awq_values = [True, False]
+    group_size_values = [1, 4]
+    ratio_values = [0.4, 0.8]
+    sensitivity_metric_values = [
+        nncf.SensitivityMetric.HESSIAN_INPUT_ACTIVATION,
+        nncf.SensitivityMetric.MAX_ACTIVATION_VARIANCE,
+    ]
+    gptq_values = [False]  # [True, False] ticket: 155538
```

Comment:
The ticket is resolved. It can be true, but I would run it once and not sure how long the test will take with it...

Reply:
Reviewed code is outdated. 40 secs is execution time for `gptq_values = [True, False]`